lib/gc.cpp

```diff
@@ -33,26 +34,57 @@ limitations under the License.
 
 // One can disable the GC, e.g., to run under Valgrind, by editing config.h
 #if HAVE_LIBGC
-static bool done_init;
+static bool done_init, started_init;
 void *operator new(std::size_t size) {
     /* DANGER -- on OSX, can't safely call the garbage collector allocation
      * routines from a static global constructor without manually initializing
      * it first.  Since we have global constructors that want to allocate
      * memory, we need to force initialization */
     if (!done_init) {
+        started_init = true;
         GC_INIT();
         done_init = true; }
     return ::operator new(size, UseGC, 0, 0);
 }
 void *operator new[](std::size_t size) {
     if (!done_init) {
+        started_init = true;
         GC_INIT();
         done_init = true; }
     return ::operator new(size, UseGC, 0, 0);
 }
 void operator delete(void *p) _GLIBCXX_USE_NOEXCEPT { return gc::operator delete(p); }
 void operator delete[](void *p) _GLIBCXX_USE_NOEXCEPT { return gc::operator delete(p); }
 
+void *realloc(void *ptr, size_t size) {
```

Comment:
Why aren't you using the glibc hooks? https://www.gnu.org/software/libc/manual/html_node/Hooks-for-Malloc.html

Reply:
Do they work on OSX libc?